prow/cmd/hmac/main.go

```diff
@@ -0,0 +1,389 @@
+
+	currentHMACMap := map[string]github.HmacsForRepo{}
+	if err := yaml.Unmarshal(currentHmacYaml, &currentHMACMap); err != nil {
+		logrus.WithError(err).Error("Couldn't unmarshal the hmac secret as hierarchical file. Parsing as a single global token and writing it back to the secret.")
+		currentHMACMap["*"] = github.HmacsForRepo{
+			github.HmacSecret{
+				Value: string(currentHmacYaml),
+			},
+		}
+	}
+
+	configAgent := &config.Agent{}
+	if err := configAgent.Start(o.configPath, ""); err != nil {
+		logrus.WithError(err).Fatal("Error starting config agent.")
+	}
+	newHMACConfig := configAgent.Config().ManagedWebhooks
+
+	c := &client{
+		kubernetesClient: kc,
+		options:          o,
+
+		currentHMACMap: currentHMACMap,
+		newHMACConfig:  newHMACConfig,
+	}
+
+	if err := c.handleConfigUpdate(); err != nil {
+		logrus.WithError(err).Fatal("Error handling hmac config update.")
+	}
+}
+
+func (c *client) handleConfigUpdate() error {
+	repoAdded := map[string]config.ManagedWebhookInfo{}
+	repoRemoved := map[string]bool{}
+	repoRotated := map[string]config.ManagedWebhookInfo{}
+
+	for repoName, hmacConfig := range c.newHMACConfig {
+		if _, ok := c.currentHMACMap[repoName]; ok {
+			repoRotated[repoName] = hmacConfig
+		} else {
+			repoAdded[repoName] = hmacConfig
+		}
+	}
+
+	for repoName := range c.currentHMACMap {
+		if _, ok := c.newHMACConfig[repoName]; !ok {
+			repoRemoved[repoName] = true
+		}
+	}
+
+	// Remove the webhooks for the given repos, as well as removing the tokens from the current hmac map.
+	if err := c.handleRemovedRepo(repoRemoved); err != nil {
+		return fmt.Errorf("error handling hmac update for removed repos: %v", err)
+	}
+
+	// Generate new hmac token for required repos, do batch update for the hmac token secret,
+	// and then iteratively update the webhook for each repo.
+	if err := c.handleAddedRepo(repoAdded); err != nil {
+		return fmt.Errorf("error handling hmac update for new repos: %v", err)
+	}
+	if err := c.handledRotatedRepo(repoRotated); err != nil {
+		return fmt.Errorf("error handling hmac rotations for the repos: %v", err)
+	}
+	// Update the hmac token secret first, to guarantee the new tokens are available to hook.
+	if err := c.updateHmacTokenSecret(); err != nil {
+		return fmt.Errorf("error updating hmac tokens: %v", err)
+	}
+	// HACK: sleep 20 seconds to wait for the reconciliation to finish.
+	time.Sleep(20 * time.Second)
+	if err := c.batchOnboardNewTokenForRepos(); err != nil {
+		return fmt.Errorf("error onboarding new token for the repos: %v", err)
+	}
+
+	// Do necessary cleanups after the token and webhook updates are done.
+	if err := c.cleanup(); err != nil {
+		return fmt.Errorf("error cleaning up %v", err)
+	}
+
+	return nil
+}
+
+// handleRemoveRepo handles webhook removal and hmac token removal from the current hmac map for all repos removed from the declarative config.
+func (c *client) handleRemovedRepo(removed map[string]bool) error {
+	repos := make([]string, len(removed))
+	i := 0
```

Comment:
0 is the default value for int, please use:

`var i int`

https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_declaration_style

Reply:
Done. And thanks for the link!